elastic/kibana
# [Infrastructure UI] Plot metric charts data based on current page items

closes [#152186](https://github.com/elastic/kibana/issues/152186)

## Summary

This PR makes the metric charts show data for the hosts on the current page. With this change, the charts will **only** load after the table has finished loading its data - or after Snapshot API has responded

It also changes the current behavior of the table pagination and sorting. Instead of relying on the `EuiInMemoryTable` the pagination and sorting are done manually, and the EuiInMemoryTable has been replaced by the `EuiBasicTable`. 

The loading indicator has also been replaced.

Paginating and sorting:

https://user-images.githubusercontent.com/2767137/233161166-2bd719e1-7259-4ecc-96a7-50493bc6c0a3.mov

Open in lens 

https://user-images.githubusercontent.com/2767137/233161134-621afd76-44b5-42ab-b58c-7f51ef944ac2.mov


### How to test
- Go to Hosts view
- Paginate and sort the table data
- Select a page size and check if the select has been stored in the localStorage (`hostsView:pageSizeSelection` key)
